Cody-Durham/ideabox-boilerplate
# Feature/local storage

### What does this PR do?
- Adds new ideas to local storage
- Deletes idea from local storage when user clicks 'X' button
- Updates star property of idea in local storage when user clicks star button

### What does it fix?
- N/A

### Is this a feature or a fix?
- feature

### Where should the reviewer start?
- idea.js lines 12-25 (new methods), and main.js lines 70 and 80

### How should this be tested?
1. open Inspector --> Application --> Local Storage
2. Add new ideas by clicking save idea button: the ideas should appear in local storage, with Keys that match their id properties
3. Refresh page: ideas should still appear in local storage
4. Delete a idea: it should disappear from local storage
5. Star poster: star property of idea in local storage should change

*can't check for locally stored idea to display in card grid after refresh, because we haven't created functionality to display local storage yet
